common_rest/src/main/java/uk/gov/dhsc/htbhf/requestcontext/aop/RequestContextAdvisor.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.dhsc.htbhf.requestcontext.aop;
+
+import lombok.RequiredArgsConstructor;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.reflect.MethodSignature;
+import uk.gov.dhsc.htbhf.requestcontext.MDCWrapper;
+import uk.gov.dhsc.htbhf.requestcontext.RequestContext;
+import uk.gov.dhsc.htbhf.requestcontext.RequestContextHolder;
+
+import java.lang.reflect.Method;
+
+import static uk.gov.dhsc.htbhf.requestcontext.MDCWrapper.REQUEST_ID_MDC_KEY;
+import static uk.gov.dhsc.htbhf.requestcontext.MDCWrapper.SESSION_ID_MDC_KEY;
+
+@RequiredArgsConstructor
+public class RequestContextAdvisor {
+
+    private final RequestContextHolder requestContextHolder;
+    private final MDCWrapper mdcWrapper;
+
+    @Around("@annotation(NewRequestContextWithSessionId)")
+    public Object updateRequestContext(ProceedingJoinPoint joinPoint) throws Throwable {
+        try {
+            String sessionId = getSessionIdFromMethodAnnotation(joinPoint);
+            String requestId = "";
+
+            RequestContext requestContext = requestContextHolder.get();
+            requestContext.setRequestId(requestId);
+            requestContext.setSessionId(sessionId);
+            mdcWrapper.put(REQUEST_ID_MDC_KEY, requestId);
+            mdcWrapper.put(SESSION_ID_MDC_KEY, sessionId);
+
+            return joinPoint.proceed();
+        } finally {
+            requestContextHolder.clear();
+            mdcWrapper.remove(REQUEST_ID_MDC_KEY);
+            mdcWrapper.remove(SESSION_ID_MDC_KEY);
```

Comment:
I know its potentially a pain to do it, but the bulk of the 10-15 lines above are duplicated in RequestIdFilter in common-rest, could/should we refactor out the common code? I'm not sold on it either way by the way, just thought I'd mention it...